### ISSUE TITLE
[TASK] Design and Implement Human-in-the-Loop Active Learning Strategy for NER

### ISSUE THREAD

**Issue by amina-mardiyyah** [ML4LitS/OTAR3088#63]
# Description

Develop a robust human-in-the-loop active learning loop tailored for Named Entity Recognition (NER)(applicable beyond the current entities for CellLine, CellType, and Tissues). 

----

## Proposed Workflow

The active learning cycle will follow these steps:

- **Initial Seed Annotation**: A small, representative subset of the unlabeled data is annotated using a generative LLM to generate initial "silver-standard" labels.

- **Expert Validation**: The LLM-generated annotations are passed to a **Christine** for review, correction, and validation, creating a quality silver standard dataset.

- **Initial Model Training**: An initial NER model (e.g., Flair or Hugging Face) is trained on this seed dataset.

- **Uncertainty Sampling**: The newly trained model performs inference on a larger pool of unlabeled data. The pipeline then identifies and samples the predictions where the model has the lowest confidence (uncertainty sampling).

- **Expert Annotation Loop**: These low-confidence predictions are presented to Christine again for validation and correction.

- **Retrain & Repeat:** The newly validated data is added to the training set, and the model is retrained. The process from Step 4 is repeated, until a satisfactory results is achieved on hidden test set.
